src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1967,6 +1967,11 @@ public TokenMetadata getTokenMetadata()
         return TokenMetadataProvider.instance.getTokenMetadata();
```

Comment:
In `getRangeToAddressMap()`, should we switch to `getTokenMetadataForKeyspace`? 

And in `ReplicaPlans#forBatchlogWrite`, `ReplicaLayout#forTokenWriteLiveAndDown`, `CFS#localRangeSplits`, `AlterTableStatement#DropCompactStorage#validateCanDropCompactStorage`

Reply:
the other usage of `StorageService#getTokenMetadata` looks safe

Reply:
@jasonstack Should it be done in BDP code first for the cases, which exist in BDP?

Reply:
Is it needed for `ReplicaPlans#forBatchlogWrite`? It is used to get local endpoint. (I might be stupid here, since I don't fully follow how it works)

Reply:
I am fixing for `ReplicaLayout#forTokenWriteLiveAndDown`, `CFS#localRangeSplits`, `AlterTableStatement#DropCompactStorage#validateCanDropCompactStorage`.

Reply:
> Is it needed for ReplicaPlans#forBatchlogWrite? It is used to get local endpoint. (I might be stupid here, since I don't fully follow how it works)

It is not fixed in BDP. The equivalent function is `computeBatchlogEndpoints` in `BatchWriter`.

Reply:
> Is it needed for ReplicaPlans#forBatchlogWrite? 

it's not needed for CNDB, as we are overwriting batchlog code in CNDB

Reply:
> In getRangeToAddressMap(), should we switch to getTokenMetadataForKeyspace?

This one is not fixed in BDP.